File: Strings/5/longSub.py
def lengthOfLongestSubstring(s):
    maxlen, n = 0, len(s)
    for i in range(n):
        l, hset = 0, set()
        for j in range(i,n):
            if s[j] not in hset:
                l += 1
                hset.add(s[j])
            else:
                break
        if l > maxlen:
            maxlen = l
    return maxlen


s = "abcabcbb"
print(lengthOfLongestSubstring(s))

File: Strings/5/longSub.js
var lengthOfLongestSubstring = function (s) {
  let maxlen= 0;
  const n = s.length;
    for (let i=0; i < n; i++) {
      let l= 0, hset = new Set();
        for (let j=i; j < n; j++) {
          if (!hset.has(s[j])) {
            l += 1;
            hset.add(s[j]);
          }
          else break
        }
        if (l > maxlen)
          maxlen = l
    }
  return maxlen
};

const s = 'abcabcbb';
console.log(lengthOfLongestSubstring(s));